inc/contract_item.class.php

```diff
@@ -205,17 +205,28 @@ static function countForContract(Contract $item) {
             continue;
          }
 
-         $query = "SELECT COUNT(*) AS cpt
-                   FROM `glpi_contracts_items`, `".$itemt->getTable()."`
-                   WHERE `glpi_contracts_items`.`contracts_id` = '".$item->getField('id')."'
-                         AND `glpi_contracts_items`.`itemtype` = '".$data['itemtype']."'
-                         AND `".$itemt->getTable()."`.`id` = `glpi_contracts_items`.`items_id`";
+         $params = [
+            'COUNT'     => 'cpt',
+            'FROM'      => self::getTable(),
+            'LEFT JOIN' => [
+               $itemt->getTable() => [
+                  self::getTable()     => 'items_id',
+                  $itemt->getTable()   => 'id'
+               ]
+            ],
+            'WHERE'     => [
+               'contracts_id' => $item->fields['id'],
+               'itemtype'     => $data['itemtype']
```

Comment:
As `$itemt` could be an Item_Devices, their tables have also `itemtype` field
For `contracts_id`, i think it's ok (better consistency btw to keep table prefix).